dspace-server-webapp/src/test/java/org/dspace/app/rest/hdlresolver/HdlResolverRestControllerIT.java

```diff
@@ -0,0 +1,171 @@
+        getClient()
+            .perform(get("/wrongController/" + publicItem1.getHandle()))
+            .andExpect(status().isNotFound());
+
+    }
+
+    /**
+     * Verifies that a null hdlIdentifier returns a
+     * 
+     * <code>HttpStatus.BAD_REQUEST</code>
+     * 
+     * @throws Exception
+     * @throws SQLException
+     * 
+     */
+    @Test
+    public void givenNullHdlIdentifierWhenCallHdlresolverThenReturnsBadRequest() throws Exception {
+
+        getClient().perform(
+                get(HdlResolverRestController.HDL_RESOLVER + "null"))
+                .andExpect(status().isBadRequest());
+        getClient().perform(
+                get(HdlResolverRestController.RESOLVE + "null"))
+                .andExpect(status().isBadRequest());
+        getClient().perform(
+                get(HdlResolverRestController.LISTHANDLES + "null"))
+                .andExpect(status().isBadRequest());
+        getClient().perform(
+                get(HdlResolverRestController.LISTPREFIXES + "null"))
+                .andExpect(status().isBadRequest());
+
+    }
+
+    /**
+     * Verifies that an empty hdlIdentifier returns a
+     * <code>HttpStatus.BAD_REQUEST</code>
+     * 
+     * @throws Exception
+     * @throws SQLException
+     * 
+     */
+    @Test
+    public void givenEmptyHdlIdentifierWhenCallHdlresolverThenReturnsNull() throws Exception {
+
+        getClient()
+            .perform(get(HdlResolverRestController.HDL_RESOLVER + " "))
+            .andExpect(status().isBadRequest());
+        getClient()
+            .perform(get(HdlResolverRestController.RESOLVE + " "))
+            .andExpect(status().isBadRequest());
+        getClient()
+            .perform(get(HdlResolverRestController.LISTHANDLES + " "))
+            .andExpect(status().isBadRequest());
+        getClient()
+            .perform(get(HdlResolverRestController.LISTPREFIXES + " "))
+            .andExpect(status().isBadRequest());
+
+    }
+
+    /**
+     * Verifies that any unmapped hdlIdentifier returns a null response
+     * 
+     * @throws Exception
+     * 
+     */
+    @Test
+    public void givenIdentifierNotMappedWhenCallHdlresolverThenReturnsNull() throws Exception {
+        getClient()
+            .perform(get(HdlResolverRestController.HDL_RESOLVER + "testHdlResolver/2"))
+            .andExpect(status().isOk()).andExpect(content().string("null"));
+        getClient()
+            .perform(get(HdlResolverRestController.RESOLVE + "testHdlResolver/2"))
+            .andExpect(status().isOk()).andExpect(content().string("null"));
+        getClient()
+            .perform(get(HdlResolverRestController.LISTHANDLES + "testHdlResolver/2"))
+            .andExpect(status().isOk()).andExpect(content().string("null"));
+        getClient()
+            .perform(get(HdlResolverRestController.LISTPREFIXES + "testHdlResolver/2"))
+            .andExpect(status().isOk()).andExpect(content().string("null"));
+    }
```

Comment:
The tests in this IT seem extremely minimal.  I'm not seeing any tests with differentiate the behaviors of `HDL_RESOLVER` vs `RESOLVE` vs `LISTHANDLES` vs `LISTPREFIXES`.  How should these endpoints work *differently*?  Currently all these tests just prove each endpoint works in the exact same way & returns the exact same response at all times (which would make me ask why do we have 4 endpoints here instead of one?).  Perhaps the REST Contract could help here, as I'd expect it to describe the differences...and then we need to ensure that tests exists to prove those differences.

Reply:
As requested by @pnbecker in this comment https://github.com/DSpace/DSpace/pull/8357#pullrequestreview-1046556703 the same rest service now responds with the same behaviour to all that endpoints.